packages/twenty-server/src/engine/core-modules/ai/services/ai-billing.service.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable, Logger } from '@nestjs/common';
+
+import { AIModelConfig } from 'src/engine/core-modules/ai/constants/ai-models.const';
+import { getAIModelById } from 'src/engine/core-modules/ai/utils/ai-model.utils';
+import { BILLING_FEATURE_USED } from 'src/engine/core-modules/billing/constants/billing-feature-used.constant';
+import { BillingMeterEventName } from 'src/engine/core-modules/billing/enums/billing-meter-event-names';
+import { BillingUsageEvent } from 'src/engine/core-modules/billing/types/billing-usage-event.type';
+import { WorkspaceEventEmitter } from 'src/engine/workspace-event-emitter/workspace-event-emitter';
+
+// Configuration: $0.001 = 1 credit
+const DOLLAR_TO_CREDIT_MULTIPLIER = 1000; // 1 / 0.001 = 1000 credits per dollar
+
+export interface TokenUsage {
+  promptTokens: number;
+  completionTokens: number;
+  totalTokens: number;
+}
+
+export interface CostCalculation {
+  costInCents: number;
+  model: AIModelConfig;
+}
+
+@Injectable()
+export class AIBillingService {
+  private readonly logger = new Logger(AIBillingService.name);
+
+  constructor(private readonly workspaceEventEmitter: WorkspaceEventEmitter) {}
+
+  async calculateCost(
+    modelId: string,
+    usage: TokenUsage,
+  ): Promise<CostCalculation> {
```

Comment:
I find the function signature a big strange. Why return the model here, I would have expected calculateCost to just return the cost, if the person wants the model they can get it with getAIModelById